### PR TITLE
Javalab: allow creation of levels in levelbuilder

### PR DESCRIPTION
**Pull request by molly-moen**
Add the option to create Javalab levels in levelbuilder, and filter by Javalab on the levelbuilder UI. Currently you can only add instructions and teacher-only markdown to a level.

## Links

- jira ticket: [CSA-143](https://codedotorg.atlassian.net/browse/CSA-143)

## Testing story
Tested locally with levelbuilder mode enabled. Also updated a unit test that verifies all the level options to include Javalab.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/scripts/levels/Allthethings Java Lab 1.level [hunk 1]
```diff
@@ -1,12 +1,16 @@
 <Javalab>
   <config><![CDATA[{
-  "game_id": 68,
-  "created_at": "2021-02-15T20:07:34.000Z",
-  "level_num": "custom",
-  "user_id": 1,
   "properties": {
+    "encrypted": "false",
+    "instructions_important": "false",
+    "short_instructions": "This is a sample Java Lab level.",
```
Comment (dmcavoy) [inline dashboard/config/scripts/levels/Allthethings Java Lab 1.level]: Please don't use short instructions for CSA. Short instructions are only for CSF style levels. You will want to use long instructions for CSA and hide the short instructions area for CSA.

Comment (dmcavoy) [reply dashboard/config/scripts/levels/Allthethings Java Lab 1.level]: Here is the line to hide short instructions : https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/levels/editors/fields/_instructions.haml#L5

Comment (dmcavoy) [reply dashboard/config/scripts/levels/Allthethings Java Lab 1.level]: You will also want to hide the short instructions area for TTS here: https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/levels/editors/fields/_tts.haml#L7-L10

**Comment by dmcavoy** [comment]
Have you asked curriculum if they want mini rubrics or the help and tips tab? 

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/views/levels/editors/fields/_instructions_area.html.haml#L7-L8

**Comment by dmcavoy** [comment]
Does curriculum plan to use Choice Levels? What about Contained Levels? You may not be ready for these but just thinking about them while I was here.

**Comment by dmcavoy** [comment]
Hi Molly - Overall this looks good. There is some special weirdness in how level editing works right now (we are hoping to fix later in the year) so I've made some suggestions to deal with that. Also offered some ideas of other things you may want to turn on because if you are using the instructions area you kinda already get them for "free"

**Comment by dmcavoy** [comment]
All of the following lines would go in you javalab.html.haml file.

To add Choice Levels:

`= render partial: 'levels/editors/fields/bubble_choice_sublevel', locals: {f: f}`

Submittable, Embed, Project Template Levels, and Contained Levels are added with:

`= render partial: 'levels/editors/fields/special_level_types', locals: {f: f}`
For this one you will need to think through when to add some of these features. I think contained levels you should be able to add with no problem. Submittable might also be something you can add with no problem. For Embed and Project Template levels there might be some work to make those work for JavaLab. If you want to just turn on some of these you will want to go into the special_level_types.html.haml file and set which ones you want.

You probably want encrypted examples. Here is the line for that

`= render partial: 'levels/editors/fields/encrypted_examples', locals: {f: f, level_type: 'javalab'}`

I don't know if editors want the ability to hide the share and remix buttons for CSA. If they do you will want:

`= render partial: 'levels/editors/fields/sharing', locals: {f: f}`
You may have to dig into that file as well as you will only want the hide share and remix buttons setting the others won't apply

You probably want callouts:

`= render partial: 'levels/editors/fields/callouts', locals: {f: f}`

**Comment by dmcavoy** [comment]
Oh one more. Audit log:

`= render partial: 'levels/editors/fields/audit_log', locals: {f: f}`

File: dashboard/app/views/levels/edit.html.haml [hunk 2]
```diff
@@ -42,6 +42,7 @@
     = render partial: 'levels/editors/spritelab', locals: {f: f} if @level.instance_of?(GamelabJr)
     = render partial: 'levels/editors/weblab', locals: {f: f} if @level.is_a? Weblab
     = render partial: 'levels/editors/ailab', locals: {f: f} if @level.is_a? Ailab
+    = render partial: 'levels/editors/javalab',locals: {f: f} if @level.is_a? Javalab
```
Comment (jmkulwik) [inline dashboard/app/views/levels/edit.html.haml]: Super NIT: missing a space after the comma `javalab',locals`

**Review by jmkulwik — COMMENTED** [review]
Deferring to Dani on the levelbuilder functionality. In general, LGTM.

**Comment by molly-moen** [comment]
Added everything Dani suggested but the special level types (submittable, contained), because it will need more than a one-line update to make it work for Javalab. Added a [backlog item](https://codedotorg.atlassian.net/browse/CSA-182) for that task.

**Review by dmcavoy — APPROVED** [review]
Looks good! Thanks for adding all those other fields!

**Pull request merged**